action.yml

```diff
@@ -41,10 +42,10 @@ runs:
       working-directory: ${{ github.action_path }}
       env:
         APPLICATION_NAME: ${{ inputs.application_name }}
+        AWS_DEFAULT_REGION: ${{ inputs.region }}
```

Comment:
I don't think it's a good idea to rename this variable.

It's still `region`, which can be set separately, with a default of `eu-central-1` - so if you change the name to `AWS_DEFAULT_REGION` this would imply for me, that this sets the default for the region (which probably will be also used for following commands) and there is another variable which is called `AWS_REGION` which would allow you to override `AWS_DEFAULT_REGION`.

I don't think that this is what you had in mind, so I would leave it as `REGION` to better reflect the meaning.

Reply:
No this is exact what I wanted to do. We can use `AWS_REGION` instead of `AWS_DEFAULT_REGION`, but we have to use the `AWS_` variables, otherwise it will not work without setting the variable in the environment section in `deploy.yml`

The workflow
```yaml
- name: Deploy
  uses: moneymeets/action-beanstalk-deploy@master
  with:
    application_name: demo-app
    environment_name: demo-environment
  env:
    AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
    AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
    AWS_REGION: eu-central-1  # or AWS_DEFAULT_REGION
```

I switched to the `AWS_` variable, because I don't want to specify the region twice.

Reply:
Yes, but you have the following code in `action.py`:

```python
region=os.environ["AWS_DEFAULT_REGION"]
```

and to my mind this would override `AWS_REGION` and it all somehow doesn't make sense.

If you want to switch to AWS variables, then I think you should remove `region` from `action.py` and `action.yml` completely, and set `AWS_DEFAULT_REGION` in `action.yml`. If users want to override it from the outside, then they could set `AWS_REGION` to do this.

Reply:
Almost, let's discuss my fixup in the meeting.